dbms/src/Core/SettingsCommon.cpp

```diff
@@ -63,6 +63,17 @@ void SettingNumber<Type>::set(const String & x)
     set(parse<Type>(x));
 }
 
+template <>
+void SettingNumber<bool>::set(const String & x)
+{
+    if (x == "false")
+        set(false);
+    else if (x == "true")
+        set(true);
+    else
+        set(parse<bool>(x));
```

Comment:
I think in this case the code becomes clear if you just make `SettingBool` a separate class instead of making a specialization like this and then define `using SettingBool = SettingNumber<bool>`.

Reply:
Don't think so. The rest methods are same with SettingNumber